### PR TITLE
add support for MCP9600 temperature sensor

### PR DESCRIPTION
**Pull request by hathach23**
I have added the new driver and datasheet for mcp9600 sensor, that uses I2C. Please help me check my code especially at the init() and read() and 3 write()-function. Thank you in advance.

### CODE REVIEW

**Comment by tridge** [comment]
pull requests need to be against master. Then we will separately consider if the change is suitable to be backported to 4.1 release after it has been accepted into master

File: ArduCopter/Copter.cpp [hunk 1]
```diff
@@ -205,6 +205,7 @@ const AP_Scheduler::Task Copter::scheduler_tasks[] = {
 #if STATS_ENABLED == ENABLED
     SCHED_TASK_CLASS(AP_Stats,             &copter.g2.stats,            update,           1, 100),
 #endif
+    SCHED_TASK(mcpUpdate, 10, 75),
```
Comment (tridge) [inline ArduCopter/Copter.cpp:208]: new sensors should be hooked into AP_Vehicle, not Copter, so they work on all vehicle types

File: libraries/AP_TemperatureSensor/AP_MCP9600.h [hunk 2]
```diff
@@ -0,0 +1,186 @@
+/*
```
Comment (tridge) [inline libraries/AP_TemperatureSensor/AP_MCP9600.h:1]: before we add this we will need to do a frontend/backend split in AP_TemperatureSensor, like we have done for other driver types, so we have a TSENS_TYPE parameter to select which type of temperature sensor to enable

**Review by tridge — CHANGES_REQUESTED** [review]
this will need a bit of work to make a frontend/backend split for temperature sensors first

**Comment by tridge** [comment]
@peterbarker we should discuss who will do the frontend/backend split, and if we do i2c simulators for these two sensor types

**Comment by tridge** [comment]
or should we stick to lua scripts for temperature? Note that sub has support for one sensor type already

**Comment by magicrub** [comment]
https://github.com/ArduPilot/ardupilot/pull/17627

**Comment by magicrub** [comment]
closed in favor of https://github.com/ArduPilot/ardupilot/pull/21662

**Pull request closed without merging**